src/gui/transferlistwidget.cpp

```diff
@@ -257,7 +257,7 @@ void TransferListWidget::setSelectedTorrentsLocation()
 
     const QString newLocation = QFileDialog::getExistingDirectory(this, tr("Choose save path"), oldLocation,
                                             QFileDialog::DontConfirmOverwrite | QFileDialog::ShowDirsOnly | QFileDialog::HideNameFilterDetails);
-    if (!QDir(newLocation).exists()) return;
+    if (newLocation.isEmpty() || !QDir(newLocation).exists()) return;
```

Comment:
And to answer my own question:
[QDir constructor](https://doc.qt.io/qt-5/qdir.html#QDir-1)
>Constructs a QDir pointing to the given directory path. If path is empty the program's working directory, ("."), is used.

Gotta love those subtleties that bite you in the ass...